### PR TITLE
Correct docker-compose demo docs

### PR DESCRIPTION
**Pull request by reify-travis-brown**
Security is enabled by default so setting this environment variable has no effect unless the intent is to disable it for demo purposes.

### Description
This change allows for a demo to be started without additional security setup. Without this change, the current instructions lead to an invalid security state that cannot be used until further security configuration is completed.

### Issues Resolved
N/A

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks, @reify-travis-brown ! 

@scrawfor99, @peterzhuamazon - Can you  approve for technical accuracy? Thanks!

**Review by scrawfor99 — CHANGES_REQUESTED** [review]
Setting `DISABLE_SECURITY_PLUGIN` to `true` will cause the Security plugin to not be used. 

It is a little confusing because it is a double negative but I think we want to keep things as they are unless I am misunderstanding the intent of this change.

File: _security/configuration/demo-configuration.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ The demo configuration is automatically called as part of the setup for each sup
 Use the following steps to set up the Security plugin using Docker:
 
 1. Download [docker-compose.yml](https://opensearch.org/downloads.html).
-2. In the `docker-compose.yml` file, set `DISABLE_SECURITY_PLUGIN` to `false`.
+2. In the `docker-compose.yml` file, set environment variable `DISABLE_SECURITY_PLUGIN` to `true`.
```
Comment (peterzhuamazon) [inline _security/configuration/demo-configuration.md:30]: This section should be changed to `if user want to disable security plugin then set to true` or similar.

More details can be seen here:
https://github.com/opensearch-project/opensearch-build/tree/main/docker/release#disable-security-plugin-security-dashboards-plugin-security-demo-configurations-and-related-configurations

We have security always enable by default so no need to specifically make it false.
And this commit of making it to true will just disable the security by default, which is not the recommended way for docker.

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]

File: _security/configuration/demo-configuration.md (hunk 1 above)
Comment (peterzhuamazon) [reply _security/configuration/demo-configuration.md:30]: The correct way to fix this is just changing `Use the following steps to set up the Security plugin using Docker` to something like `Use the following steps to disable Security plugin when using Docker`, or just directly link to the readme file for docker release in build repo:
https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/README.md#disable-security-plugin-security-dashboards-plugin-security-demo-configurations-and-related-configurations

We do not want to recommend users to disable security unless they specifically want to do so.

Comment (hdhalter) [reply _security/configuration/demo-configuration.md]: @peterzhuamazon - Let me know if this works. I took that line out of the process steps and moved it below.

File: _security/configuration/demo-configuration.md [hunk 2]
```diff
@@ -27,14 +27,15 @@ The demo configuration is automatically called as part of the setup for each sup
...
-2. In the `docker-compose.yml` file, set `DISABLE_SECURITY_PLUGIN` to `false`.
-3. Run the following command:
+2. Run the following command:
 
 ```bash
 docker-compose up
 ```
 {% include copy.html %}
 
+If you want to disable the Security plugin when using Docker, set the `DISABLE_SECURITY_PLUGIN` environment variable  to `true` in the `docker-compose.yml` file. Disabling the Security plugin is not recommended. For more information, see the [Docker release notes](https://github.com/opensearch-project/opensearch-build/tree/main/docker/release#disable-security-plugin-security-dashboards-plugin-security-demo-configurations-and-related-configurations) in GitHub.
```
Comment (peterzhuamazon) [inline _security/configuration/demo-configuration.md:37]: Hi @reify-travis-brown could you rename `Docker release notes` to `Docker image release README`?

Thanks.

Comment (hdhalter) [inline _security/configuration/demo-configuration.md:37]:
```suggestion
If you want to disable the Security plugin when using Docker, set the `DISABLE_SECURITY_PLUGIN` environment variable  to `true` in the `docker-compose.yml` file. Disabling the Security plugin is not recommended. For more information, see the [Docker image release README](https://github.com/opensearch-project/opensearch-build/tree/main/docker/release#disable-security-plugin-security-dashboards-plugin-security-demo-configurations-and-related-configurations) in GitHub.
```

**Review by peterzhuamazon — APPROVED** [review]

**Review by scrawfor99 — APPROVED** [review]

**Pull request merged**